builder/rewrite.h

```diff
@@ -315,7 +315,7 @@ SLINKY_UNIQUE bool match(const pattern_select<C, T, F>& p,
 
 template <typename C, typename T, typename F>
 SLINKY_UNIQUE expr substitute(const pattern_select<C, T, F>& p, const match_context& ctx) {
-  return select::make(substitute(p.c, ctx), substitute(p.t, ctx), substitute(p.f, ctx));
+  return select::make(expr(substitute(p.c, ctx)), expr(substitute(p.t, ctx)), expr(substitute(p.f, ctx)));
```

Comment:
I'm a bit confused why does it need ```expr()``` around ```substitute()```, isn't it already returns expr()?

Reply:
This overload does, but other overloads don't (or didn't). In particular, `substitute(pattern_wildcard<N>, ...)` returned `base_expr_node*`. However, now it returns `expr_ref`, so these explicit conversions aren't necessary any more.